topics/go/algorithms/slices/max/max.go

```diff
@@ -1,24 +1,21 @@
 package slices
 
-// Max returns the maximum integer in the slice.
-func Max(n []int) int {
+import "fmt"
 
-	// If the length of the slice is 1 then return the integer at index 0.
-	if len(n) == 1 {
-		return n[0]
+// Max returns the maximum integer in the slice.
+func Max(n []int) (max int, err error) {
+	if len(n) == 0 {
+		return 0, fmt.Errorf("slice %#v has no elements", n)
 	}
 
-	// Create an integer and assign it to the first index of the slice.
-	max := n[0]
-
 	// Loop over the slice of integers.
-	for _, num := range n {
-
-		// If num is greater than max. Assign max to num.
-		if num > max {
+	for i, num := range n {
+		// if this is our first iteration or if num is greater than max. Assign max
```

Comment:
I need a line space above every comment. We need proper sentence structure and punctuation.

Reply:
done.